pkg/kp/pcstore/consul_store.go

```diff
@@ -154,6 +157,76 @@ func (s *consulStore) Delete(id fields.ID) error {
 	return s.applicator.RemoveAllLabels(labels.PC, id.String())
 }
 
+// performs a safe (ie check-and-set) mutation of the pc with the given id,
+// using the given function
+// if the mutator returns an error, it will be propagated out
+// if the returned PC has id="", then it will be deleted
+func (s *consulStore) mutatePc(
+	id fields.ID,
+	mutator func(fields.PodCluster) (fields.PodCluster, error),
+	modifyIndex uint64,
```

Comment:
sorry for further churn, but you should get the `modifyIndex` from your `Get()` call below, then the caller doesn't need to fetch the pod cluster beforehand at all!